runelite-client/src/main/java/net/runelite/client/plugins/config/SearchablePluginComparator.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2020, Jack Hodkinson <https://github.com/jrhodkinson>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.config;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.Function;
+import org.apache.commons.lang3.StringUtils;
+import static java.util.Comparator.comparing;
+import static java.util.Comparator.naturalOrder;
+import static java.util.Comparator.nullsLast;
+import static java.util.Comparator.reverseOrder;
+
+public class SearchablePluginComparator implements Comparator<SearchablePlugin<?>>
+{
+
+	private final String query;
+	private final List<String> queryPieces;
+
+	public SearchablePluginComparator(String query)
+	{
+		this.query = query.trim().toLowerCase();
+		this.queryPieces = Arrays.asList(this.query.split(" "));
+	}
+
+	@Override
+	public int compare(SearchablePlugin<?> plugin1, SearchablePlugin<?> plugin2)
+	{
+		return comparator().compare(plugin1, plugin2);
+	}
+
+	public Comparator<SearchablePlugin<?>> comparator()
+	{
+		if (StringUtils.isBlank(query))
+		{
+			return nullsLast(isPinned()).thenComparing(name());
+		}
+		return nullsLast(exactMatch())
```

Comment:
the comparator should be cached in a final field. Most of these methods are single use and should be inlined (+ a comment if its not otherwise obvious what they do)